### DISCUSSION TITLE
Html as entry point

### DISCUSSION THREAD

**Discussion by else1fy** [web-infra-dev/rsbuild#3522] [General]
Hello, tell me, are you planning to add the possibility of using html as an entry point?

This was a fairly popular [topic](https://github.com/webpack/webpack/issues/536) for discussion, and even a special [plugin](https://github.com/webdiscus/html-bundler-webpack-plugin) for webpack was created (which unfortunately does not work in rspack), as well, this is what [vite can](https://vitejs.dev/guide/build.html#multi-page-app) do out of the box

**Comment by AlexanderOpran** [comment]
Any updates regarding this? I can try to make a PR if someone can point me in the right direction on where to start.

**Reply by 9aoy** [reply]
The [html-bundler-webpack-plugin](https://github.com/webdiscus/html-bundler-webpack-plugin) can achieve this functionality, but it seems that it is not yet supported for use with Rspack. You might consider integrating `html-bundler-webpack-plugin` into Rspack. If you encounter any issues, feel free to open an issue in the [Rspack](https://github.com/web-infra-dev/rspack/issues) repository, and we will provide support as soon as possible.

**Reply by PointSingularity** [reply]
I kinda have a roundabout solution, but it might not be exactly what you are looking for (I use an almost empty HTML template, and have the script as a file next to it):

```ts
import { defineConfig } from '@rsbuild/core';
import { isDev } from '../constants.mjs';

export default defineConfig({
  source: {
    entry: {
      index: 'src/iframes/example.ts',
    },
    decorators: {
      version: 'legacy',
    },
  },
  output: {
    distPath: {
      root: `${isDev ? 'build' : 'dist'}/html/`,
    },
    filename: {
      html: 'example.html',
    },
    injectStyles: true,
    inlineScripts: true,
    legalComments: 'none',
    target: 'web',
  },
  tools: {
    swc: {
      jsc: {
        parser: {
          syntax: 'typescript',
          tsx: false,
          decorators: true,
        },
        transform: {
          legacyDecorator: true,
          decoratorMetadata: true,
        },
        target: 'es5',
      },
      env: undefined,
    },
    htmlPlugin: {
      template: 'src/iframes/example.html',
      inject: 'head',
      title: undefined,
      meta: {},
    },
  },
  performance: {
    chunkSplit: {
      strategy: 'all-in-one',
    },
  },
});
```

Which then produces one file:
```
❯ yarn html
[html:prod]   Rsbuild v1.2.11
[html:prod] 
[html:dev]   Rsbuild v1.2.11
[html:dev] 
[html:prod] ready   Built in 0.08 s (web)
[html:dev] ready   Built in 0.08 s (web)
[html:prod] 
[html:prod]   File (web)                Size       Gzip    
[html:prod]   dist/html/example.html    12.1 kB    3.8 kB
[html:prod] 
[html:dev] 
[html:dev]   File (web)                 Size       Gzip    
[html:dev]   build/html/example.html    12.1 kB    3.8 kB
[html:dev] 
[html:prod] yarn run html:prod exited with code 0
[html:dev] yarn run html:dev exited with code 0
```
